src/wwwroot/js/genpage/gentab/currentimagehandler.js

```diff
@@ -392,12 +395,15 @@ function shiftToNextImagePreview(next = true, expand = false) {
     let newIndex = index + (next ? 1 : -1);
     if (newIndex < 0) {
         newIndex = imgs.length - 1;
-    }
-    else if (newIndex >= imgs.length) {
+    } else if (newIndex >= imgs.length) {
         newIndex = 0;
     }
     let newImg = imgs[newIndex];
     let block = findParentOfClass(newImg, 'image-block');
+    for (let block of getRequiredElementById('current_image_batch').querySelectorAll('.image-block-selected')) {
```

Comment:
I think setting the selection here is redundant if `setCurrentImage` is going to replace it right after anyway?